### PR TITLE
Copy 2020/2021 vision code

### PR DESCRIPTION
**Pull request by OliverW10**
copies a lot of the code for managing cameras, networking and deploying from vision-2020
I modified vision.py to use this but have left the core of the vision for the recruits to do

also added a requirements.txt (which somehow we didnt have last year) and a bit more of a readme

deploy.py and sim.py were slightly simplified from last year

### CODE REVIEW

**Review by james-ward — COMMENTED** [review]
lgtm, but I'd like @auscompgeek to do the final approval - I've been away from this code for two years!

File: README.md [hunk 1]
```diff
@@ -1,2 +1,29 @@
 # vision-2022
-The Drop Bears' vision code for FIRST Infinite Recharge (FRC 2022)
+The Drop Bears' vision code for FIRST Rapid React (FRC 2022)
+
+## Setup
+To install dependencies do:
+
+    pip install -r requirments.txt
```
Comment (auscompgeek) [inline README.md:7]:
```suggestion
    pip install -r requirements.txt
```

File: camera_manager.py [hunk 2]
```diff
@@ -1,14 +1,200 @@
...
+
+    def notify_error(self, error: str) -> None:
+        """Prints an error to the console.
+        Args:
+            error: The string to print.
+        """
+        print(error, file=sys.stderr)
+
+    def set_camera_property(self, property, value) -> None:
+        ...
```
Comment (auscompgeek) [inline camera_manager.py:123]: `...` indicates to me that the method should be implemented in future (but isn't now and doing nothing in the time being is okay). Will these be implemented, or should they just be `pass`?

**Review by auscompgeek — APPROVED** [review]
lgtm with comments

File: camera_manager.py (hunk 2 above)
Comment (james-ward) [reply camera_manager.py:123]: Good catch - we want this code to actually be runnable, even if it doesn't do what it needs to.

I agree with Davo - these should be `pass`

**Pull request merged**